labs/src/main/scala/org/scalalabs/basic/lab02/CollectionExercise.scala

```diff
@@ -33,7 +33,30 @@ object CollectionExercise01 {
    *
    */
   def googleCodeJamGooglerese(lines: String*): Seq[String] = {
-    error("fix me")
+      //figure out missing character mapping
+    val input = "ejp mysljylc kd kxveddknmc re jsicpdrysi rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd de kr kd eoya kw aej tysr re ujdr lkgc jv " filterNot (_ == ' ')
+    val output = "our language is impossible to understand there are twenty six factorial possibilities so it is okay if you want to just give up" filterNot (_ == ' ')
+
+    val alphabet = 'a' to 'z'
+    val missingIn = alphabet filterNot (input contains _)
+    alphabet diff input
+
+    val missingOut = alphabet filterNot (output contains _)
+    alphabet diff output
+    //z and q, hint says z -> q, so remaining is: q -> z
+
+    //visualize missing chars in alphabet
+    val existingCharsSorted = input.toSet.toList.sorted.mkString
+    val visualMissingChars = alphabet.map(c => if (existingCharsSorted.contains(c)) c else ' ').mkString
+
+    //compute mapping
+    val initialMapping = (input zip output).toSet
+    //ensure 1 to 1 mapping
+    initialMapping.groupBy(_._1).values.forall(_.size == 1)
+
+    val mapper = Map('z' -> 'q', 'q' -> 'z', ' ' -> ' ').withDefaultValue('?') ++ initialMapping
+
+    lines.map(_ map mapper)
```

Comment:
You did this much more elegantly than my version. I simply encoded the conversions in a map -- very manual and inferior to this way. Great work!